packages/backend/src/lib/hasuraClient.ts

```diff
@@ -3,11 +3,24 @@ import { GraphQLClient } from 'graphql-request';
 import { CONFIG } from '../config';
 import { getSdk } from './autogen/hasura-sdk';
 
-export const client = getSdk(
-  new GraphQLClient(CONFIG.graphqlURL, {
-    headers: {
-      'Content-Type': 'application/json',
-      'x-hasura-access-key': CONFIG.adminKey,
-    },
-  }),
-);
+interface GetClientParams {
+  role?: string;
+  userId?: string;
+  backendOnly?: boolean;
+}
+
+export const getClient = (params: GetClientParams = {}) =>
+  getSdk(
+    new GraphQLClient(CONFIG.graphqlURL, {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-hasura-access-key': CONFIG.adminKey,
+        'x-hasura-role': params.role || 'admin',
+        'x-hasura-user-id': params.userId || '',
+        'x-hasura-use-backend-only-permissions':
+          params.backendOnly === true ? 'true' : 'false',
+      },
```

Comment:
Are these new params used anywhere right now?

Reply:
I was using it, then abandoned as I mentioned in the PR description https://github.com/MetaFam/TheGame/pull/340#issue-571268272

Not used anymore, but maybe it will be, do you want to keep/remove that ?